### PR TITLE
libressl: update to 3.4.2

### PR DESCRIPTION
**Pull request by artkiver**
#### Description
Fixed: https://trac.macports.org/ticket/63615 

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [x] security fix

###### Tested on

macOS 12.0.1 21A559 arm64
Xcode 13.1 13A1030d

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by artkiver** [comment]
This is hopefully a cleaner attempt, with a single commit rather than the previous PR #13237.

Per @reneeotten 's comments. I do not profess to be a git rebase wizard and given that my attempts to drop previous version commit history only added to the commits, it seemed as if starting over would be simpler?

Additionally, while I did ostensibly attempt "tried existing tests with sudo port test?" I left that unchecked as there do not appear to be any test.run parameters specified as per portfile(7) resulting in the following output, which is presumably, extraneous:

"Testing libressl                                   
Error: Failed to test libressl: libressl has no tests turned on. see 'test.run' in portfile(7)
Error: See /opt/local/var/macports/logs/_Users_grey_src_githubclones_mpblah_macports-ports_security_libressl/libressl/main.log for details.
Error: Follow https://guide.macports.org/#project.tickets if you believe there
is a bug."

Regardless, hopefully, this attempt is closer to acceptable? Thanks!

**Comment by herbygillot** [comment]
So far this looks good. Let's wait for the maintainer (@jeremyhu) to review.

**Review by mascguy — APPROVED** [review]

**Pull request merged**
